crates/rome_js_analyze/src/semantic_analyzers/style/no_static_only_class.rs

```diff
@@ -0,0 +1,164 @@
+use crate::semantic_services::Semantic;
+use rome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_js_syntax::{AnyJsClass, AnyJsClassMember};
+use rome_rowan::{AstNode, AstNodeList};
+
+declare_rule! {
+    /// Succinct description of the rule.
+    ///
+    /// Put context and details about the rule.
+    /// As a starting point, you can take the description of the corresponding _ESLint_ rule (if any).
+    ///
+    /// Try to stay consistent with the descriptions of implemented rules.
+    ///
+    /// Add a link to the corresponding ESLint rule (if any):
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/rule-name
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// class X {
+    ///   static foo = false;
+    ///   static bar() {};
+    /// }
+    /// ```
+    /// ```js,expect_diagnostic
+    /// class StaticConstants {
+    ///   static readonly version = 42;
+    ///
+    ///   static isProduction() {
+    ///     return process.env.NODE_ENV === 'production';
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// const X = {
+    ///   foo: false,
+    ///   bar() {}
+    /// };
+    /// ```
+    /// ```js
+    /// export const version = 42;
+    ///
+    /// export function isProduction() {
+    ///   return process.env.NODE_ENV === 'production';
+    /// }
+    ///
+    /// function logHelloWorld() {
+    ///   console.log('Hello, world!');
+    /// }
+    /// ```
+    ///
+    pub(crate) NoStaticOnlyClass {
+        version: "next",
+        name: "noStaticOnlyClass",
+        recommended: false,
+    }
+}
+
+impl Rule for NoStaticOnlyClass {
+    type Query = Semantic<AnyJsClass>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let class_declaration = ctx.query();
+
+        // If the class has decorators, we can't be sure that the suer can use a module-based approach instead
+        if class_declaration.decorators().len() > 0 {
+            return None;
+        }
+
+        let all_members_static =
+            class_declaration
+                .members()
+                .into_iter()
+                .all(|member| match member {
+                    // TODO: Clean up this mess
+                    AnyJsClassMember::JsBogusMember(_) => true, // TODO: What is this?
```

Comment:
```suggestion
                    AnyJsClassMember::JsBogusMember(_) => false,
```

This highlights an error in the source. If an error occurs we should not emit a diagnostic (i.e. None should be return). False is thus more adequate here.